lib/compiler/src/v3_core.erl

```diff
@@ -1837,22 +1846,13 @@ generator(Line, {generate,Lg,P0,E}, Gs, St0) ->
     {Head,St1} = list_gen_pattern(P0, Line, St0),
     {[Tail,Skip],St2} = new_vars(2, St1),
     {Cg,St3} = lc_guard_tests(Gs, St2),
-    {AccPat,SkipPat} = case Head of
-                           #c_var{} ->
-                               %% If the generator pattern is a variable, the
-                               %% pattern from the accumulator clause can be
-                               %% reused in the skip one. lc_tq and bc_tq1 takes
-                               %% care of dismissing the latter in that case.
-                               Cons = ann_c_cons(LA, Head, Tail),
-                               {Cons,Cons};
-                           nomatch ->
-                               %% If it never matches, there is no need for
-                               %% an accumulator clause.
-                               {nomatch,ann_c_cons(LA, Skip, Tail)};
-                           _ ->
-                               {ann_c_cons(LA, Head, Tail),
-                                ann_c_cons(LA, Skip, Tail)}
-                       end,
+    AccPat = case Head of
+                 nomatch ->
+                     nomatch;
+                 _ ->
+                     ann_c_cons(LA, Head, Tail)
+             end,
+    SkipPat = ann_c_cons(LA, Skip, Tail),
```

Comment:
If this is a refactoring, I am not sure that this is correct.
Before, `#c_var{}` case was returning `{ann_c_cons(LA, Head, Tail), ann_c_cons(LA, Head, Tail)}`.
Now, the case for a `#c_var{}` returns `{ann_c_cons(LA, Head, Tail), ann_c_cons(LA, Skip, Tail)}`, where the difference is in the change of `Head` by `Skip`.

If this is intended, please disregard this comment

Reply:
It is an intentional simplification. The variable name in the head of the skip clause doesn't matter, because it is never used.